service/src/main/java/uk/gov/hmcts/reform/fpl/service/casesubmission/CaseSubmissionTemplateDataGenerationService.java

```diff
@@ -0,0 +1,741 @@
+            StringBuilder sb = new StringBuilder(document.getDocumentStatus());
+            if (!equalsIgnoreCase(document.getDocumentStatus(), ATTACHED.getLabel())
+                && StringUtils.isNotEmpty(document.getStatusReason())) {
+                sb.append(NEW_LINE).append(document.getStatusReason());
+            }
+            return sb.toString();
+
+        }
+        return DEFAULT_STRING;
+    }
+
+    private List<DocmosisSocialWorkOther> formatAnnexDocumentDisplay(
+        final List<Element<DocumentSocialWorkOther>> otherSocialWorkDocuments) {
+        return otherSocialWorkDocuments.stream()
+            .map(Element::getValue)
+            .filter(Objects::nonNull)
+            .map(this::buildDocmosisSocialWorkOther)
+            .collect(toList());
+    }
+
+    private DocmosisSocialWorkOther buildDocmosisSocialWorkOther(DocumentSocialWorkOther document) {
+        return DocmosisSocialWorkOther.builder()
+            .documentTitle(
+                StringUtils.isNotEmpty(document.getDocumentTitle()) ? document.getDocumentTitle() : DEFAULT_STRING)
+            .build();
+    }
+
+    private DocmosisHearingPreferences buildDocmosisHearingPreferences(
+        final HearingPreferences hearingPreferences) {
+        final boolean hearingPreferencesPresent = (hearingPreferences != null);
+        return DocmosisHearingPreferences.builder()
+            .interpreter(hearingPreferencesPresent
+                ? concatenateKeyAndValue(
+                hearingPreferences.getInterpreter(),
+                hearingPreferences.getInterpreterDetails()) : DEFAULT_STRING)
+            .welshDetails(hearingPreferencesPresent
+                ? concatenateKeyAndValue(
+                hearingPreferences.getWelsh(),
+                hearingPreferences.getWelshDetails()) : DEFAULT_STRING)
+            .intermediary(hearingPreferencesPresent
+                ? concatenateKeyAndValue(
+                hearingPreferences.getIntermediary(),
+                hearingPreferences.getIntermediaryDetails()) : DEFAULT_STRING)
+            .disabilityAssistance(hearingPreferencesPresent
+                ? concatenateKeyAndValue(
+                hearingPreferences.getDisabilityAssistance(),
+                hearingPreferences.getDisabilityAssistanceDetails()) : DEFAULT_STRING)
+            .extraSecurityMeasures(hearingPreferencesPresent
+                ? concatenateKeyAndValue(
+                hearingPreferences.getExtraSecurityMeasures(),
+                hearingPreferences.getExtraSecurityMeasuresDetails()) : DEFAULT_STRING)
+            .somethingElse(hearingPreferencesPresent
+                ? concatenateKeyAndValue(
+                hearingPreferences.getSomethingElse(),
+                hearingPreferences.getSomethingElseDetails()) : DEFAULT_STRING)
+            .build();
+    }
+
+    private DocmosisRisks buildDocmosisRisks(final Risks risks) {
+        final boolean risksPresent = (risks != null);
```

Comment:
same

Reply:
Same for `buildDocmosisFactorsParenting` and `buildDocmosisInternationalElement`